### PR TITLE
Gradle Arquillian Adapter Improvements

### PR DESCRIPTION
**Pull request by roguexz**
This commit allows developers to make use of the upcoming Gradle 5 features around importing Maven BOM dependency constraints.

* Make use of the default Gradle tooling model to determine the project dependencies, as compared to scanning the command line.
* Verified the plugin's capability on a multi-module Gradle project.

- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [ ] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

**Comment by roguexz** [comment]
Raised a PR for the Thorntail examples as well.

**Comment by roguexz** [comment]
Looks like the [Jenkins instance](https://ci.wildfly-swarm.io/) is not accessible again, so can't say if the linux-build job kicked off or not.

File: arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java
Comment (Ladicek) [inline arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java:67]: Is it guaranteed that there's always an `IdeaProject`, or does that depend on the project having the IntelliJ plugin?

File: arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java
Comment (Ladicek) [inline arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java]: Isn't `parallelStream` a bit excessive here?

File: arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java
Comment (roguexz) [reply arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java:67]: I have tested on the following scenarios and it has been working well.

1. All the Gradle projects in the examples repository. None of them have the IntelliJ plugin on them.
2. In my personal project (large multi-module project) which also doesn't have the IntelliJ plugin in it.
3. Trying out different configurations: compile, provided, testCompile.


I need to raise another change that will help improve compatibility with the new [`java-library`](https://docs.gradle.org/current/userguide/java_library_plugin.html) opinion of Gradle. This would support for configurations like `api`, `testImplementation`.

File: arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java
Comment (roguexz) [reply arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java]: Sorry about that. I was trying out different modes to see if I can speed up the execution time. Let me get some timing numbers and confirm if it would benefit us or not.

File: arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java
Comment (roguexz) [inline arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java:70]: @Ladicek Fixed the issue. I thought I was iterating over the dependencies .. didn't realize that I was iterating over the projects. Thanks for pointing it out :)

File: arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java
Comment (roguexz) [reply arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java:67]: Just verified the existing changes with the `java-library` plugin and things are working without any new modifications.

**Comment by sberyozkin** [comment]
retest this please

**Comment by sberyozkin** [comment]
Not sure it will make it to 2.2.1 with the build to start shortly. I'll wait for the approval from Ladislav anyway.
I see you plan more updates so hopefully targeting it for 2.3.0 should work for you as well 
thanks.

**Comment by roguexz** [comment]
Sounds good. Though having this commit would make the integration "logically complete" for supporting Gradle 4+ projects. The next set of changes are around performance and simplifying the build script definitions.

Either way is fine by me.

**Comment by roguexz** [comment]
@Ladicek Any further comments before this PR gets merged?

File: arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java
Comment (Ladicek) [reply arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java:70]: I didnt' really point out anything, but glad you found a problem and fixed it :-)

**Comment by Ladicek** [comment]
I don't really have anything, I don't understand Gradle at all (I did know a _little_ bit about it back in the Gradle 1.0 RC days, but that's a _long_ time ago) :-) I find using the IDEA project model slightly weird, but that's not a blocker or anything like that.

**Review by sberyozkin — APPROVED** [review]

**Pull request merged**
